setup.py

```diff
@@ -18,9 +18,10 @@
 
 install_requires = [
     'enum34; python_version < "3.4"',
-    'google-auth-oauthlib>=0.0.1,<1.0.0',
-    'google-api-core[grpc] >= 1.1.0, < 2.0.0dev',
-    'PyYAML >=4.2b1, < 5.0',
+    'google-auth-oauthlib >= 0.0.1, < 1.0.0',
+    'google-api-core == 1.7.0',
+    'grpcio == 1.18.0',
```

Comment:
Regarding pinning: This is discouraged for libraries rather than end-user code, as it means your users may be unable to get important bug/security fixes in newer versions of your dependencies, and your users will run into dependency issues if they also want use other libraries that ever require newer versions of your transitive dependencies.

Reply:
Resolving this for this PR but will revisit in a separate PR.